main_test.go

```diff
@@ -0,0 +1,55 @@
+package main
```

Comment:
I may have missed it, but do we run tests in the CI currently?

Reply:
At this time there is no CI for the Go code. I have written the GHA workflows for that as part of another PR. Though, it may make more sense to add them here. Let me know if you want me to do that.

Reply:
I think it's fine that it is in another PR